src/passes/implicitConversionToExplicit.ts

```diff
@@ -257,34 +256,42 @@ export class ImplicitConversionToExplicit extends ASTMapper {
 }
 
 function insertConversionIfNecessary(expression: Expression, targetType: TypeNode, ast: AST): void {
-  const currentType = generalizeType(getNodeType(expression, ast.compilerVersion))[0];
-  targetType = generalizeType(targetType)[0];
+  const [currentType, currentLoc] = generalizeType(getNodeType(expression, ast.compilerVersion));
+  const generalisedTargetType = generalizeType(targetType)[0];
 
   if (currentType instanceof AddressType) {
-    if (!(targetType instanceof AddressType)) {
-      insertConversion(expression, targetType, ast);
+    if (!(generalisedTargetType instanceof AddressType)) {
+      insertConversion(expression, generalisedTargetType, ast);
     }
   } else if (currentType instanceof ArrayType) {
     assert(
-      targetType instanceof ArrayType,
+      generalisedTargetType instanceof ArrayType,
       `Unable to convert array ${printNode(expression)} to non-array type ${printTypeNode(
-        targetType,
+        generalisedTargetType,
       )}`,
     );
-    const elementT = targetType.elementT;
-    if (expression instanceof TupleExpression && expression.isInlineArray) {
-      expression.vOriginalComponents.forEach((element) => {
-        assert(element !== null, `Unexpected empty slot in inline array ${printNode(expression)}`);
-        insertConversionIfNecessary(element, elementT, ast);
-      });
-      expression.typeString = generateExpressionTypeString(
-        specializeType(targetType, DataLocation.Memory),
-      );
+    if (currentLoc === DataLocation.Memory) {
+      const parent = expression.parent;
+      const [replacement, shouldReplace] = ast
+        .getUtilFuncGen(expression)
+        .memory.convert.genIfNecesary(expression, targetType);
```

Comment:
Just noting that there are now two memory.convert in the codebase. One here and one in dataAccessFunctionalizer. I am guessing there are edge cases for both?

Reply:
Yep, this one is the general case one, but I lfet the other one in for safety and for if anything gets introduced between the passes. As they only gen if necessary, it shouldn't ever lead to a double conversion

Reply:
OK cool cool